lib/puppet/provider/package_targetable.rb

```diff
@@ -25,14 +25,14 @@
 class Puppet::Provider::Package::Targetable < Puppet::Provider::Package
   # Prefetch our package list, yo.
   def self.prefetch(packages)
-    catalog_packages = packages.first[1]::catalog::resources.select{ |p| p.provider.class == self }
+    catalog_packages = packages.first[1]::catalog::resources.select{ |p| p.provider.class.to_s == self.to_s }
```

Comment:
@branan Could you peek at the change here, precipitated by the addition of the spec test (and maybe look at the test itself) below? 😬

Reply:
Why is it necessary to compare `to_s`?

Reply:
Simply because I cannot otherwise define a test where the code considers the classes equal.
Which I why I asked for help :)

Reply:
@joshcooper to re-evaluate/help :)

Reply:
ah the problem is the spec test defines another `gem` provider, which causes us to have 2 provider classes named `Puppet::Type::Package::ProviderGem` but different object ids. So change:

```diff
-  let(:provider) { Puppet::Type.type(:package).provide(:gem, :parent => described_class) }
+  let(:provider) { Puppet::Type.type(:package).provider(:gem) }
```

as the `gem` provider is already targetable

Reply:
Also a couple of changes requested:

1. Please use dots instead of double colons when calling methods, such as `packages.first[1].catalog.resources`.
2. The call to `packages.first[1]` makes it seem like `packages` is an Array of Arrays, but really it's a Hash where the key is the resource name and the value is the resource. It also causes the Hash values to be copied into a new array each time it's called:
  ```
  (byebug) packages.first.object_id
  70177932340900
  (byebug) packages.first.object_id
  70177932358240
  ```
Recommend calling `packages.values.first` instead

  ```
(byebug) packages.values.first.object_id
70178598557700
(byebug) packages.values.first.object_id
70178598557700
```

Reply:
Oh, nice. Thx!